pallets/liquidity-pools-gateway/src/tests.rs

```diff
@@ -259,7 +326,247 @@ mod remove_instance {
 
-mod process_msg {
+mod process_msg_axelar_relay {
+	use sp_core::bounded::BoundedVec;
+
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(SOURCE_CHAIN_EVM_ID, SOURCE_ADDRESS);
+			let relayer_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(LiquidityPoolsGateway::add_instance(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert_ok!(LiquidityPoolsGateway::add_relayer(
+				RuntimeOrigin::root(),
+				relayer_address.clone(),
+			));
+
+			let expected_msg = MessageMock::First;
+			let expected_domain_address = domain_address.clone();
+
+			let mut msg = Vec::new();
+			msg.extend_from_slice(&(LENGTH_SOURCE_CHAIN as u32).to_be_bytes());
+			msg.extend_from_slice(&SOURCE_CHAIN);
+			msg.extend_from_slice(&(LENGTH_SOURCE_ADDRESS as u32).to_be_bytes());
+			msg.extend_from_slice(&SOURCE_ADDRESS);
+			msg.extend_from_slice(&expected_msg.serialize());
+
+			MockLiquidityPools::mock_submit(move |domain, message| {
+				assert_eq!(domain, expected_domain_address);
+				assert_eq!(message, expected_msg);
+				Ok(())
+			});
+
+			let expected_domain_address = domain_address.clone();
+
+			MockOriginRecovery::mock_try_convert(move |origin| {
+				let (source_chain, source_address) = origin;
+
+				assert_eq!(&source_chain, SOURCE_CHAIN.as_slice());
+				assert_eq!(&source_address, SOURCE_ADDRESS.as_slice());
+
+				Ok(expected_domain_address.clone())
+			});
+
+			assert_ok!(LiquidityPoolsGateway::process_msg(
+				GatewayOrigin::AxelarRelay(relayer_address).into(),
+				BoundedVec::<u8, MaxIncomingMessageSize>::try_from(msg).unwrap()
+			));
+		});
+	}
+
+	#[test]
+	fn invalid_message_origin() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::Centrifuge(get_test_account_id().into());
+			let relayer_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(LiquidityPoolsGateway::add_relayer(
+				RuntimeOrigin::root(),
+				relayer_address.clone(),
+			));
+
+			let expected_msg = MessageMock::First;
+
+			let mut msg = Vec::new();
+			// Need to prepend length signaler
+			msg.extend_from_slice(&(0 as u32).to_be_bytes());
+			msg.extend_from_slice(&(0 as u32).to_be_bytes());
+			msg.extend_from_slice(&expected_msg.serialize());
+
+			MockOriginRecovery::mock_try_convert(move |origin| {
+				let (source_chain, source_address) = origin;
+
+				assert!(source_chain.is_empty());
+				assert!(source_address.is_empty());
+
+				Ok(domain_address.clone())
+			});
+
+			assert_noop!(
+				LiquidityPoolsGateway::process_msg(
+					GatewayOrigin::AxelarRelay(relayer_address).into(),
+					BoundedVec::<u8, MaxIncomingMessageSize>::try_from(msg).unwrap()
+				),
+				Error::<Runtime>::InvalidMessageOrigin,
+			);
+		});
+	}
+
+	#[test]
+	fn unknown_instance() {
```

Comment:
Out of curiosity: Do I understand correctly this test fails because we do not `mock_submit` as in the the `success` test? 

Nit: To me, it is not trivial to spot the error. A short comment would be great.

Reply:
No. I it fails as the `Allowlist` storage is not set for the given sender.